sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -974,6 +974,28 @@ def get_cmis_media_lanes_mask(self, api, appl, lport, subport):
 
         return media_lanes_mask
 
+    def reset_app_code_and_reinit_all_lanes(self, api):
+        """
+	   Rest and DP init all the lanes
+        """
+        api.set_application(0xff, 0, 0)
+        api.set_datapath_deinit(0xff)
+        if not api.scs_apply_datapath_init(0xff):
```

Comment:
@AnoopKamath this is NOT required


![image](https://github.com/sonic-net/sonic-platform-daemons/assets/45705344/832adc75-0ae4-4dbc-900c-f77ecfb6dab0)

Reply:
Hi @prgeor,
We should apply (apply is applyDPInit) on all the lanes to unused(appsel 0), if we are setting different app code to partial lanes. When 800G config is active (all lanes used), then applying partial config (only one port of 400G) is clearly “partial” and “invalid” which is exactly that the module is responding.
Host must either apply both 400G ports or apply NULL config (to clear all lanes) before applying one port of 400G. DPDeactivated state and set app will push config- appsel 0 to Stage Control Set and we need to apply DPInit to push it to Active Control Set so that all the lanes become unused. Just doing DPDeinit and setting app to 0 does not solve the problem.

![image](https://github.com/sonic-net/sonic-platform-daemons/assets/115578705/a04f945a-e998-42c0-9729-e9a99f8fc95e)